lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -8830,31 +8830,38 @@ namespace Js
     {
         if (this->GetState() != CleanedUp)
         {
-            // Unregister xdataInfo before OnCleanup() which may release xdataInfo->address
```

Comment:
>OnCleanup [](start = 43, length = 9)

Why not just delete the table her if we are shutting down already?  Wouldn't the background thread be shutdown by now anyway?

Reply:
this is the major code path causing the hang. and no, the background thread was not shutdown yet at this point, the background thread was shutdown along with destroying the thread context. (nativeCodeGen is shutdown before this but the thread is still alive).